magenta/reviews/GAN.md

```diff
@@ -0,0 +1,136 @@
+<img src="assets/gan/image12.png"> is very close to zero and the generator cannot learn quickly.
+This can be fixed by maximizing <img src="assets/gan/image11.png"> instead, which converges to the same solution, but
+provides strong gradients when <img src="assets/gan/image03.png"> is low.
+
+Here’s a visual explanation of how the learning happens.
+
+<p align="center">
+ <img src="assets/gan/image09.png"><br>
+ <i>Training a GAN to fit a data distribution.</i>
+</p>
+
+Imagine that the real data is represented by some distribution, drawn as a dotted black line on the figure.
+Real examples then are obtained by sampling from it.
+<img src="assets/gan/image04.png"> is the domain of the model's input.
+Fake examples are obtained by sampling from some distribution on domain <img src="assets/gan/image02.png">.
+The generator maps this <img src="assets/gan/image02.png"> to <img src="assets/gan/image04.png"> which results in fake examples,
+whicha are distributed according to the green line on the figure.
+The dashed blue line is the discriminator and it gives each example from <img src="assets/gan/image04.png"> a probability of
+being real.
+
+Starting with a fixed, random generator and a discriminator (see Figure (a)), we begin by training the discriminator
+to distinguish real examples from fake examples produced by the generator (see Figure (b)).
+Once trained, we fix the discriminator and train the generator to maximize discriminator’s error (Figure (c)).
+Then we train the discriminator again, and so forth, until convergence.
+When it converges, the generator has learned a true data distribution which matches the black dotted line exactly.
+Therefore the discriminator has no way of telling whether an example is real or fake, it returns ½ on all of them
+(Figure (d)).
+
+Under certain conditions, this process converges to a fixed point, where the generator has learned the true
+data distribution, and hence the discriminator cannot classify real examples from generated ones.
+The original paper’s model is capable of generating great results on simple tasks such as MNIST.
+Recent advances have made it possible to get high quality examples on more complicated problems, like generating faces and CIFAR
+images.
+
+## Experiments.
+
+<p align="center">
+ <img src="assets/gan/image05.png"><br>
+</p>
+
+To evaluate model performance quantitatively, one could calculate the likelihood of the real examples given the
+distribution that the model induces.
+MNIST examples generated by GANs have approximated log likelihood 225 with standard error 2 across several
+dataset folds.
+In the same setting Deep Belief Network, which is an undirected graphical model, generates examples with approximated log likelihood 138
+with similar error.
+Examples generated from Toronto Face Dataset, which is a much harder problem, have approximated log likelihood of
```

Comment:
I mean that we cannot calculate the true log likelihood, so we approximate in somehow.